strawberryfields/decompositions.py

```diff
@@ -136,24 +138,66 @@ def graph_embed(A, max_mean_photon=1.0, make_traceless=True, tol=1e-6):
     if m != n:
         raise ValueError("The matrix is not square.")
 
-    if np.linalg.norm(A-np.transpose(A)) >= tol:
+    if not np.allclose(A, np.transpose(A), rtol=rtol, atol=atol):
         raise ValueError("The matrix is not symmetric.")
 
     if make_traceless:
-        A = A - np.trace(A)*np.identity(n)/n
+        A = A - np.trace(A) * np.identity(n) / n
 
-    s, U = takagi(A, tol=tol)
-    sc = np.sqrt(1.0+1.0/max_mean_photon)
-    vals = -np.arctanh(s/(s[0]*sc))
+    s, U = takagi(A, tol=atol)
+    sc = np.sqrt(1.0 + 1.0 / max_mean_photon)
+    vals = -np.arctanh(s / (s[0] * sc))
+    return vals, U
+
+
+def graph_embed(A, mean_photon=1.0, make_traceless=False, atol=1e-08):
+    r"""Embed a graph into a Gaussian state.
+
+    Given a graph in terms of a symmetric adjacency matrix
+    (in general with arbitrary complex off-diagonal and real diagonal entries),
+    returns the squeezing parameters and interferometer necessary for
+    creating the Gaussian state whose off-diagonal parts are proportional to that matrix.
+
+    Uses :func:`takagi`.
+
+    Args:
+        A (array[complex]): square, symmetric (weighted) adjacency matrix of the graph
+        max_mean_photon (float): It guarantees that the mean photon number in the pure Gaussian state
```

Comment:
Since this is the docstring of `graph_embed`, we should be implicit:
```suggestion
        max_mean_photon (float): guarantees that the mean photon number in the pure Gaussian state
```

Reply:
For graph embed the second argument should not be called max_mean_photon since there is nothing maximum about; it is something that you set and that is not greater than any other quantity.